http-common/src/proxy.rs

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::env;
+use std::future::Future;
+use std::io;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
+
+pub enum MaybeProxyStream<C> {
+    NoProxy(C),
+    Proxy(hyper_proxy::ProxyStream<C>),
+}
+
+impl<C> AsyncRead for MaybeProxyStream<C>
+where
+    C: AsyncRead + AsyncWrite + Unpin,
+{
+    fn poll_read(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        buf: &mut ReadBuf<'_>,
+    ) -> Poll<io::Result<()>> {
+        match &mut *self {
+            MaybeProxyStream::NoProxy(s) => Pin::new(s).poll_read(cx, buf),
+            MaybeProxyStream::Proxy(s) => Pin::new(s).poll_read(cx, buf),
+        }
+    }
+}
+
+impl<C> AsyncWrite for MaybeProxyStream<C>
+where
+    C: AsyncRead + AsyncWrite + Unpin,
+{
+    fn poll_write(
+        mut self: Pin<&mut Self>,
+        ctx: &mut Context<'_>,
+        buf: &[u8],
+    ) -> Poll<io::Result<usize>> {
+        match &mut *self {
+            MaybeProxyStream::NoProxy(s) => Pin::new(s).poll_write(ctx, buf),
+            MaybeProxyStream::Proxy(s) => Pin::new(s).poll_write(ctx, buf),
+        }
+    }
+
+    fn poll_flush(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        match &mut *self {
+            MaybeProxyStream::NoProxy(s) => Pin::new(s).poll_flush(ctx),
+            MaybeProxyStream::Proxy(s) => Pin::new(s).poll_flush(ctx),
+        }
+    }
+
+    fn poll_shutdown(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<()>> {
+        match &mut *self {
+            MaybeProxyStream::NoProxy(s) => Pin::new(s).poll_shutdown(ctx),
+            MaybeProxyStream::Proxy(s) => Pin::new(s).poll_shutdown(ctx),
+        }
+    }
+}
```

Comment:
Add `poll_write_vectored` and `is_vectored`

Reply:
Fixed.